### PR TITLE
Add table aws_guardduty_ipset. Closes #240

### LINKED ISSUE #240: Add table aws_guardduty_ipset
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.38.1/service/guardduty/guarddutyiface

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_guardduty_ipset []

PRETEST: tests/aws_guardduty_ipset

TEST: tests/aws_guardduty_ipset
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_guardduty_detector.primary: Creating...
aws_s3_bucket.bucket: Creating...
aws_guardduty_detector.primary: Still creating... [10s elapsed]
aws_s3_bucket.bucket: Still creating... [10s elapsed]
aws_guardduty_detector.primary: Creation complete after 10s [id=16bc36ca7122bf855a3db64ef0752b4a]
aws_s3_bucket.bucket: Still creating... [20s elapsed]
aws_s3_bucket.bucket: Still creating... [30s elapsed]
aws_s3_bucket.bucket: Still creating... [40s elapsed]
aws_s3_bucket.bucket: Still creating... [50s elapsed]
aws_s3_bucket.bucket: Still creating... [1m0s elapsed]
aws_s3_bucket.bucket: Creation complete after 1m2s [id=terraform-20210326080935793600000001]
aws_s3_bucket_object.MyIPSet: Creating...
aws_s3_bucket_object.MyIPSet: Creation complete after 7s [id=MyIPSet]
aws_guardduty_ipset.named_test_resource: Creating...
aws_guardduty_ipset.named_test_resource: Still creating... [10s elapsed]
aws_guardduty_ipset.named_test_resource: Still creating... [20s elapsed]
aws_guardduty_ipset.named_test_resource: Creation complete after 25s [id=16bc36ca7122bf855a3db64ef0752b4a:b0bc36caf8f2822cd9d8b3792ada4ba6]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
detector_id = 16bc36ca7122bf855a3db64ef0752b4a
ipset_id = b0bc36caf8f2822cd9d8b3792ada4ba6
resource_aka = arn:aws:guardduty:us-east-1:986325076436:detector/16bc36ca7122bf855a3db64ef0752b4a/ipset/b0bc36caf8f2822cd9d8b3792ada4ba6
resource_name = turbottest28677

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "format": "TXT",
    "name": "turbottest28677",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest28677"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "account_id": "986325076436",
    "format": "TXT",
    "name": "turbottest28677",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest28677"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "format": "TXT",
    "name": "turbottest28677",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest28677"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:guardduty:us-east-1:986325076436:detector/16bc36ca7122bf855a3db64ef0752b4a/ipset/b0bc36caf8f2822cd9d8b3792ada4ba6"
    ],
    "title": "turbottest28677"
  }
]
✔ PASSED

POSTTEST: tests/aws_guardduty_ipset

TEARDOWN: tests/aws_guardduty_ipset

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  detector_id,
  ipset_id,
  name,
  format,
  location
from
  aws_guardduty_ipset;

+----------------------------------+----------------------------------+--------------+--------+------------------------------------------------------------------------------
| detector_id                      | ipset_id                         | name         | format | location                                                                     
+----------------------------------+----------------------------------+--------------+--------+------------------------------------------------------------------------------
| 40bc3701ed3f8a9331a0f7cd2582ae50 | 60bc3702a015e7ebec584ed484a034ba | test-ipset1  | TXT    | https://s3.amazonaws.com/aws-logs-986325076436-us-east-1/customtrustlist.csv 
| 28bc3705b369a9f655994633edcbfd16 | 0abc3707f0eb9ec7cc185c55ed0f288f | test-ipset22 | TXT    | https://s3.amazonaws.com/aws-logs-986325076436-us-east-2/customtrustlist.csv 
+----------------------------------+----------------------------------+--------------+--------+------------------------------------------------------------------------------
```

```
select
  ipset_id,
  name,
  status
from
  aws_guardduty_ipset
where
  status = 'INACTIVE';

+----------------------------------+------+----------+
| ipset_id                         | name | status   |
+----------------------------------+------+----------+
| 0ebc5576f1ac67c688c395f84f43b656 | test | INACTIVE |
+----------------------------------+------+----------+
```
</details>

### CODE REVIEW

File: aws/table_aws_guardduty_ipset.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+				Description: "The status of IPSet.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsGuardDutyIPSet,
+			},
+			{
+				Name:        "location",
+				Description: "The URI of the file that contains the IPSet.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsGuardDutyIPSet,
+			},
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:64]:
```suggestion
			},
			
```

File: aws/table_aws_guardduty_ipset.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+func tableAwsGuarDutyIPSet(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_guardduty_ipset",
+		Description: "AWS GuarDuty IPSet",
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:16]:
```suggestion
		Description: "AWS GuardDuty IPSet",
```

File: aws/table_aws_guardduty_ipset.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"detector_id", "ipset_id"}),
+			ShouldIgnoreError: isNotFoundError([]string{"InvalidInputException", "NoSuchEntityException", "BadRequestException"}),
+			Hydrate:           getAwsGuardDutyIPSet,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listGuardDutyDetectors,
+			Hydrate:       listAwsGuarDutyIPSets,
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:24]:
```suggestion
			Hydrate:       listAwsGuardDutyIPSets,
```

File: aws/table_aws_guardduty_ipset.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+type ipsetInfo = struct {
+	guardduty.GetIPSetOutput
+	IPSetID    string
+	DetectorID string
+}
+
+//// LIST FUNCTION
+
+func listAwsGuarDutyIPSets(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:98]:
```suggestion
func listAwsGuardDutyIPSets(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: docs/tables/aws_guardduty_ipset.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  ipset_id,
+  name,
+  format,
+  location
+from
+  aws_guardduty_ipset;
+```
+
+
+### List ipsets which are active
```
Comment (Paulami30) [inline docs/tables/aws_guardduty_ipset.md:21]:
```suggestion
### List IPSets which are not active
```

File: aws/table_aws_guardduty_ipset.go [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+				Transform:   transform.FromField("DetectorID"),
+			},
+			{
+				Name:        "ipset_id",
+				Description: "The ID of the IPSet.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("IPSetID"),
+			},
+			{
+				Name:        "name",
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:42]: Make this the first column

File: aws/table_aws_guardduty_ipset.go [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+			}
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:133]:
```suggestion
//// HYDRATE FUNCTION
```

File: aws/table_aws_guardduty_ipset.go [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+	// Get call
+	data, err := svc.GetIPSet(params)
+	if err != nil {
+		return nil, err
+	}
+
+	return ipsetInfo{*data, id, detectorID}, nil
+}
+
+//// TRANSFORM FUNCTIONS
```
Comment (Paulami30) [inline aws/table_aws_guardduty_ipset.go:176]:
```suggestion
//// TRANSFORM FUNCTIONS
```
```suggestion
//// TRANSFORM FUNCTION
```

**Review by Paulami30 — COMMENTED** [review]
Please check the comments and resolve merge conflict

File: aws/service.go [hunk 9]
```diff
@@ -1034,3 +1035,26 @@ func EventBridgeService(ctx context.Context, d *plugin.QueryData, region string)
...
+	// so it was not in cache - create service
+	sess, err := getSession(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+	svc := guardduty.New(sess)
+	d.ConnectionManager.Cache.Set(serviceCacheKey, svc)
+
+	return svc, nil
+}
```
Comment (Paulami30) [inline aws/service.go:1060]: sort the functions. Should be after GlacierService

**Review by Paulami30 — APPROVED** [review]

**Comment by Paulami30** [comment]
LGTM

File: aws/table_aws_guardduty_ipset.go [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/guardduty"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+func tableAwsGuarDutyIPSet(_ context.Context) *plugin.Table {
```
Comment (cbruno10) [inline aws/table_aws_guardduty_ipset.go:13]: This is misspelled, should be `GuardDuty`, not `GuarDuty`.

File: aws/plugin.go [hunk 11]
```diff
@@ -86,6 +86,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_eventbridge_rule":                   tableAwsEventBridgeRule(ctx),
 			"aws_glacier_vault":                      tableAwsGlacierVault(ctx),
 			"aws_guardduty_detector":                 tableAwsGuardDutyDetector(ctx),
+			"aws_guardduty_ipset":                    tableAwsGuarDutyIPSet(ctx),
```
Comment (cbruno10) [inline aws/plugin.go:89]: This is misspelled, should be `GuardDuty`, not `GuarDuty`.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**